tests/test_csc.py

```diff
@@ -170,6 +170,24 @@ def setUp(self):
             await task
             await self.check_exposureState(harness.remote, ExposureState.DONE)
 
+            # Check that out of range durations do not put us in FAULT,
+            # and do not change the exposure state.
+            duration = 1e-9  # seconds
+            task = asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
+                                                                           duration=duration))
+            # No ExposureState message should have been emitted.
+            with self.assertRaises(asyncio.TimeoutError):
+                await harness.remote.evt_exposureState.next(flush=False, timeout=STD_TIMEOUT)
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED,
+                                             result_contains="Exposure duration"):
+                await task
```

Comment:
This is fine, but you could simplify it a bit by starting with the assertRaisesAckError and I'm wondering if you had a strong reason for your order. To be clear, this is the simpler version I had in mind:
```
with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED,
                                 result_contains="Exposure duration"):
    asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
                                                            duration=duration))
# No ExposureState message should have been emitted.
with self.assertRaises(asyncio.TimeoutError):
    await harness.remote.evt_exposureState.next(flush=False, timeout=STD_TIMEOUT)
```
this also eliminates the need test for the next "No ExposureState message should have been emitted" (immediately below).

Reply:
But the thing that raises is when we `await task`, not when we create the task.

I can get rid of one of the ExposureState checks, that's true.

Reply:
Oops. I wrote it wrong. I meant to suggest:
```
with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED,
                                 result_contains="Exposure duration"):
    await harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
                                                            duration=duration)
# No ExposureState message should have been emitted.
with self.assertRaises(asyncio.TimeoutError):
    await harness.remote.evt_exposureState.next(flush=False, timeout=STD_TIMEOUT)```

Reply:
(We had a back and forth on this but I don't see it any more so I'll amend my comment)